lighthouse-core/gather/driver.js

```diff
@@ -702,13 +702,23 @@ function captureJSCallUsage(funcRef, set) {
       // First frame is the function we injected (the one that just threw).
       // Second, is the actual callsite of the funcRef we're after.
       const callFrame = structStackTrace[1];
-      const file = callFrame.getFileName();
+      let url = callFrame.getFileName();
       const line = callFrame.getLineNumber();
       const col = callFrame.getColumnNumber();
-      // TODO: add back when we want stack traces. See https://github.com/GoogleChrome/lighthouse/issues/957
-      // const stackTrace = structStackTrace.slice(1).map(
-      //    callsite => callsite.toString());
-      return {url: file, args, line, col}; // return value is e.stack
+      const isEval = callFrame.isEval();
+      const stackTrace = structStackTrace.slice(1).map(callsite => callsite.toString());
+
+      // If we don't have an URL, (e.g. eval'd code), use the last entry in the
```

Comment:
could you also give an example here as to what the url will be? I know you could piece it together from the v8 API, but it'd be nice to get a sense for what it's getting at here too :)